### DISCUSSION TITLE
[CHORE] How to add approval permissions in Elsa 3.1.3 designer?

### DISCUSSION THREAD

**Discussion by shxylgithub** [elsa-workflows/elsa-core#5722] [Q&A]
How to add approval permissions to nodes in Elsa 3.1.3 designer? For example, the first level approval is for the team leader, the second level approval is for the department manager, and the third level approval is for the general manager.
![approval](https://github.com/elsa-workflows/elsa-core/assets/171897414/40c7cd1f-7b85-474a-a132-73fc812977d8)

**Comment by glime-ai** [comment]
To add approval permissions to nodes in Elsa 3.1.3 designer, you would typically use a combination of user roles and activities that enforce these roles at different stages of the workflow. Here's a general approach to achieve this:

1. **Define User Roles**: Define roles in your system such as "Team Leader", "Department Manager", and "General Manager". These roles should be associated with the user accounts in your identity management system.

2. **Create Workflow**: Use the Elsa Designer to create a workflow that represents the approval process. You can use activities like "ReceiveHttpRequest" to listen for incoming approval requests.

3. **Add Approval Activities**: For each level of approval, add an activity that represents the approval action. This could be a custom activity that you create, which checks the role of the user attempting the approval.

4. **Enforce Roles**: Within each approval activity, enforce the permission check to ensure that only users with the appropriate role can execute the activity. This might involve integrating with your identity management system to verify the user's role.

5. **Set Up Notifications**: Use activities like "SendEmail" to notify approvers when their approval is needed. You can configure these notifications to be sent to users based on their roles.

6. **Handle Approvals**: When an approver performs an approval action, the workflow should record this and move to the next stage of the process, potentially notifying the next approver.

7. **Visualize Workflow**: The Elsa Designer allows you to visualize the workflow, showing the different stages of approval and the roles associated with each stage.

Unfortunately, the image link you provided does not work, so I cannot give you specific guidance based on that image. However, the general steps above should guide you in setting up a multi-level approval workflow with permissions.

If you need to create custom activities or have specific requirements for integrating with an identity management system, you may need to write some code to handle these aspects. Elsa's documentation and community can provide further guidance on how to achieve this.

Please note that as of my knowledge cutoff date in April 2023, there may have been updates or changes to Elsa Workflows. It is always a good idea to refer to the latest documentation or seek assistance from the Elsa community for the most current information.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
